relm4/src/factory/sync/collections/hashmap.rs

```diff
@@ -48,12 +49,105 @@ where
     }
 }
 
+#[derive(Debug)]
+pub struct FactoryHashMapBuilder<K, C: FactoryComponent> {
+    widget: C::ParentWidget,
+    _key: PhantomData<K>,
+}
+
+impl<K, C> FactoryHashMapBuilder<K, C>
+where
+    C: FactoryComponent,
+{
+    /// Creates a new [`FactoryHashMapBuilder`].
+    #[must_use]
+    pub fn new(widget: C::ParentWidget) -> Self {
+        Self {
+            widget,
+            _key: PhantomData,
+        }
+    }
+
+    pub fn launch(self) -> FactoryHashMapConnector<K, C> {
+        let (output_sender, output_receiver) = crate::channel();
+
+        FactoryHashMapConnector {
+            widget: self.widget,
+            output_sender,
+            output_receiver,
+            hasher: RandomState::default(),
+            _key: self._key,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct FactoryHashMapConnector<K, C, S = RandomState>
+where
+    C: FactoryComponent,
+{
+    widget: C::ParentWidget,
+    output_sender: Sender<C::Output>,
+    output_receiver: Receiver<C::Output>,
+    hasher: S,
+    _key: PhantomData<K>,
+}
+
+impl<K, C> FactoryHashMapConnector<K, C>
+where
+    C: FactoryComponent,
+{
+    pub fn forward<F, Msg>(self, f: F, forward_sender: Sender<Msg>) -> FactoryHashMap<K, C>
```

Comment:
Currently, all `forward` methods on connectors seem to use `&Sender<Msg>` as parameter and it would be nice to keep this consistent.

Reply:
Agreed - I've updated this with that change. (Also I reordered the sender and the transformer arguments to match the order used in the FactoryVecDeque and Component connectors.)